contracts/L2ChainInfo.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+/*
+This contract lives on L2 and shares information about the chain.
+It needs to get this information by being called after a fork.
+We made it for the ForkArbitrator to get the result after a fork. 
+Other contracts may also find it useful.
+It must be called after a fork until it's updated.
+Queries against it will revert until the update is done.
+
+TODO: Should we somehow make it updateable even if you miss the fork window?
```

Comment:
what is the fork window?

Reply:
By this I mean if you fork over dispute 1, then 1 week later you fork again over dispute 2, and you don't call the update to L2ChainInfo during that 1 week after the previous fork but before the new one, L2ChainInfo will never get the result of dispute 1.

Reply:
OK, I still need to make some tests for it but I've made it so that you can update the L2ChainInfo either for your current ForkingManager or for any ancestor to that ForkingManager. So the L2 should always be able to get data about disputes, even if there was some weird availability issue that means they don't get the memo until after some subsequent fork.